server/src/main/java/org/opensearch/indices/replication/SegmentReplicationState.java

```diff
@@ -119,57 +225,303 @@ protected void validateAndSetStage(Stage expected, Stage next) {
                 "can't move replication to stage [" + next + "]. current stage: [" + stage + "] (expected [" + expected + "])"
             );
         }
-        stopTimersAndSetStage(next);
-    }
-
-    private void stopTimersAndSetStage(Stage next) {
-        // save the timing data for the current step
-        stageTimer.stop();
-        timingData.add(new Tuple<>(stage.name(), stageTimer.time()));
-        // restart the step timer
-        stageTimer.reset();
-        stageTimer.start();
         stage = next;
     }
 
     public void setStage(Stage stage) {
         switch (stage) {
             case INIT:
                 this.stage = Stage.INIT;
+                overallTimer.reset();
```

Comment:
Would prefer we kept our single timingData object that tracks the time of each stage vs having 6 separate timers.

Reply:
Sure I will refactor this